File: python/findindex.py
class Solution:
	def strStr(self,source,target):
		if source is None or target is None:
			return -1
		
		for i in range(len(source)-len(target)+1):
			for j in range(len(target)):
				if source[i+j] != target[j]:
					break
				else:
					return i
		return -1

source = input("please input you source String:")
target = input("please input you target String:")

anson = Solution()

num = anson.strStr(source,target);

print("Output:",num)

File: JAVA/findindex.java
import java.util.Scanner;

public class findindex{

	public static int strStr(String source,String target)
	{
		if(source == null || target == null) return -1;

		for(int i =0;i<source.length()-target.length()+1;i++)
		{
			int j =0;
			for(;j<target.length();j++)
			{
			    if(source.charAt(i+j) != target.charAt(j)) break;
			}
			if(j == target.length()) return i;
		}
		return -1;
	}

	public static void main(String args[])
	{	
		int num;
		String source = null;
		String target = null;

		System.out.println("please input your source:");
		Scanner source_temp = new Scanner(System.in);
		source = source_temp.next();

		System.out.println("please input you target:");
		Scanner target_temp = new Scanner(System.in);
		target = target_temp.next();

		num = strStr(source,target);
		System.out.println("Output:");
		System.out.println(num);
	}

}
